src/Microsoft.VisualStudio.Sdk.TestFramework/GlobalServiceProvider.cs

```diff
@@ -136,6 +146,18 @@ int OLE.Interop.IServiceProvider.QueryService(ref Guid guidService, ref Guid rii
             IntPtr pUnk = IntPtr.Zero;
             try
             {
+                object? service = serviceOrFactory switch
+                {
+                    Func<Task<object>> factory => this.joinableTaskContext.Factory.Run(factory),
+                    _ => serviceOrFactory,
+                };
+
+                if (service is null)
+                {
+                    ppvObject = IntPtr.Zero;
+                    return VSConstants.E_NOINTERFACE;
+                }
+
                 pUnk = Marshal.GetIUnknownForObject(service);
```

Comment:
unrelated to this change, but this leaks the CCW. GetIUnknownForObject calls AddRef, QueryInterface calls AddRef again. The caller is responsible for calling Release on the returned IntPtr, but you'd need to call Release on the pUnk since you created that AddRef

Reply:
But I _do_ call release on the `pUnk` in a `finally` block below here.